processor_function/vm_porter/runner.py

```diff
@@ -188,11 +189,19 @@ def read_env_vars():
         "vmss_msi_id": os.environ.get('VMSS_MSI_ID', None),
 
         # Needed for running porter
+        "arm_use_msi": os.environ["ARM_USE_MSI"],
         "arm_subscription_id": os.environ['ARM_SUBSCRIPTION_ID'],
         "arm_client_id": os.environ["ARM_CLIENT_ID"],
-        "arm_client_secret": os.environ["ARM_CLIENT_SECRET"],
-        "arm_tenant_id": os.environ["ARM_TENANT_ID"],
+        "arm_tenant_id": os.environ["ARM_TENANT_ID"]
     }
+
+    if env_vars["arm_use_msi"] == "false":
```

Comment:
env_vars["arm_client_secret"]  = os.environ["ARM_CLIENT_SECRET"] if env_vars["arm_use_msi"] == "false" else "not_required"

Reply:
trying to get head round that - so thats a single line? any docs/python 101 I can read?!